python/lsst/ip/isr/linearize.py

```diff
@@ -26,226 +26,510 @@
+                Number of amplifiers considered.
+            ``"numLinearized"``
+                Number of amplifiers linearized.
+        """
 
+    def fromDetector(self, detector):
+        """Read linearity parameters from a detector.
 
-class LinearizeBase(metaclass=abc.ABCMeta):
-    """Abstract base class functor for correcting non-linearity
+        Parameters
+        ----------
+        detector : `lsst.afw.cameraGeom.detector`
+            Input detector with parameters to use.
+        """
+        self._detectorName = detector.getName()
+        self._detectorSerial = detector.getSerial()
+        self.populated = True
+
+        # Do not translate Threshold, Maximum, Units.
+        for amp in detector.getAmplifiers():
+            ampName = amp.getName()
+            self.linearityCoeffs[ampName] = amp.getLinearityCoeffs()
+            self.linearityType[ampName] = amp.getLinearityType()
+            self.linearityBBox[ampName] = amp.getBBox()
+
+    def fromYaml(self, yamlObject):
+        """Read linearity parameters from a dict.
+
+        Parameters
+        ----------
+        yamlObject : `dict`
+            Dictionary containing detector and amplifier information.
+        """
+        self._detectorName = yamlObject['detectorName']
+        self._detectorSerial = yamlObject['detectorSerial']
+        self.populated = True
+        self.override = True
+
+        for amp in yamlObject['amplifiers']:
+            ampName = amp['name']
+            self.linearityCoeffs[ampName] = amp.get('linearityCoeffs', None)
+            self.linearityType[ampName] = amp.get('linearityType', 'None')
+            self.linearityBBox[ampName] = amp.get('linearityBBox', None)
+
+    def toDict(self):
+        """Return linearity parameters as a dict.
+
+        Returns
+        -------
+        outDict : `dict`:
+        """
+        outDict = {'detectorName': self._detectorName,
+                   'detectorSerial': self._detectorSerial,
+                   'hasTable': self._table is not None,
+                   'amplifiers': dict()}
+        for ampName in self.linearityType:
+            outDict['amplifiers'][ampName] = {'linearityType': self.linearityType[ampName],
+                                              'linearityCoeffs': self.linearityCoeffs[ampName],
+                                              'linearityBBox': self.linearityBBox[ampName]}
+
+    def getLinearityTypeByName(self, linearityTypeName):
+        """Determine the linearity class to use from the type name.
+
+        Parameters
+        ----------
+        linearityTypeName : str
+            String name of the linearity type that is needed.
+
+        Returns
+        -------
+        linearityType : `~lsst.ip.isr.linearize.LinearizeBase`
+            The appropriate linearity class to use.  If no matching class
+            is found, `None` is returned.
+        """
+        for t in [LinearizeLookupTable,
+                  LinearizeSquared,
```

Comment:
Probably not, but I suspect it would require a second RFC to remove it.  This ensures that currently working code doesn't break, and no camera definitions need changing either.